pkg/operator/controller/dns_status_test.go

```diff
@@ -212,142 +225,204 @@ func TestComputeDNSDegradedCondition(t *testing.T) {
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 4, intstr.FromInt(1)),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.FromInt(1)),
-			expected:     operatorv1.ConditionTrue,
-		},
-		{
-			name:         "node-resolver pods unavailable is ok (integer)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 6, intstr.FromInt(1)),
-			nrDaemonset:  makeDaemonSet(6, 4, intstr.FromInt(1)),
-			expected:     operatorv1.ConditionFalse,
-		},
-		{
-			name:         "enough available (percentage)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(100, 90, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(100, 67, intstr.FromString("33%")),
-			expected:     operatorv1.ConditionFalse,
-		},
-		{
-			name:         "enough available (integer)",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 5, intstr.FromInt(1)),
-			nrDaemonset:  makeDaemonSet(6, 5, intstr.FromInt(1)),
-			expected:     operatorv1.ConditionFalse,
-		},
-		{
-			name:         "all available",
-			clusterIP:    "172.30.0.10",
-			dnsDaemonset: makeDaemonSet(6, 6, intstr.FromString("10%")),
-			nrDaemonset:  makeDaemonSet(6, 6, intstr.FromString("33%")),
-			expected:     operatorv1.ConditionFalse,
+			name:              "0 available, DNS invalid MaxUnavailable",
+			clusterIP:         "172.30.0.10",
+			dnsDaemonset:      makeDaemonSet(6, 0, intstr.FromString("TEST")),
+			oldCondition:      defaultOldCondition,
+			oldGraceCondition: defaultOldGraceCondition,
+			expected:          operatorv1.ConditionTrue,
+			expectRequeue:     false,
+		},
+		{
+			name:              "DNS invalid MaxUnavailable (string with digits without a percent sign)",
+			clusterIP:         "172.30.0.10",
+			dnsDaemonset:      makeDaemonSet(6, 6, intstr.IntOrString{Type: intstr.String, StrVal: "10"}),
+			oldCondition:      defaultOldCondition,
+			oldGraceCondition: defaultOldGraceCondition,
+			expected:          operatorv1.ConditionTrue,
+		},
+		{
+			name:              "DNS invalid MaxUnavailable (string with letters)",
+			clusterIP:         "172.30.0.10",
+			dnsDaemonset:      makeDaemonSet(6, 6, intstr.IntOrString{Type: intstr.String, StrVal: "TEST"}),
+			oldCondition:      defaultOldCondition,
+			oldGraceCondition: defaultOldGraceCondition,
+			expected:          operatorv1.ConditionTrue,
+		},
+		{
+			name:              "no clusterIP, 0 available",
+			clusterIP:         "",
+			dnsDaemonset:      makeDaemonSet(6, 0, intstr.FromString("10%")),
+			oldCondition:      defaultOldCondition,
+			oldGraceCondition: defaultOldGraceCondition,
+			expectRequeue:     false,
+			expected:          operatorv1.ConditionTrue,
+		},
+		{
+			name:              "no clusterIP, DNS invalid MaxUnavailable",
+			clusterIP:         "",
+			dnsDaemonset:      makeDaemonSet(6, 6, intstr.FromString("TEST")),
+			oldCondition:      defaultOldCondition,
+			oldGraceCondition: defaultOldGraceCondition,
+			expected:          operatorv1.ConditionTrue,
```

Comment:
Was this an intentional change (from reporting "Unknown")?  I suppose this is all right—it simplifies the logic, and you could make the case that the operator is degraded when the user provides invalid input that prevents accurate status reporting.

Reply:
IIrc, you asked me to set the DNSInvalidDNSMaxUnavailable as ConditionTrue in https://github.com/openshift/cluster-dns-operator/pull/290#discussion_r715758605

Reply:
I was saying you should set `InvalidDNSMaxUnavailable=True`, but not necessarily `Degraded=True`.  However, if you want to say `Degraded=True` when the user provides an invalid value for `MaxUnavailable`, that doesn't seem too unreasonable.